workshopapps/pictureminer.api
# feat(BE-62): download csv endpoint

## [Linear ticket](https://linear.app/team-discripto/issue/BE-62/download-csv-endpoint)

## Task: 
create an endpoint to download a csv file containing all images in a particular batch.

## Details:
When initiated, it gets the batch ID from the request, fetch all the images which belongs to the ID, transform to csv and return the csv file.
format of the csv: url, tag
## How to test: 
- If running locally or live, vist the enpoint (http://localhost:9000/api/v1/csv-service/download/:batchid)
- ```batchid``` represents a valid batch ID such as ```638b61d037a213abb85cb2fe```

- A csv file of the format url, tag should immediately be downloaded to your device. 
P.s when testing live, visit (https://discripto.hng.tech/api1/api/v1/csv-service/download/638b61d037a213abb85cb2fe)

###Preview:
<img width="1280" alt="Screenshot 2022-12-03 at 17 11 55" src="https://user-images.githubusercontent.com/86890896/205455872-8ade9eeb-d472-43bf-99a8-4e08551b5489.png">
---------------------------------------
<img width="1280" alt="Screenshot 2022-12-03 at 17 12 07" src="https://user-images.githubusercontent.com/86890896/205455884-267d78f3-6841-4ee8-82f5-00cfee485926.png">

Preview of csv
<img width="1280" alt="Screenshot 2022-12-03 at 17 12 21" src="https://user-images.githubusercontent.com/86890896/205455907-3f17bd1e-0e67-43eb-be32-d2a5dd07998e.png">
